sdk/avalon_sdk/ethereum/ethereum_worker_registry.py

```diff
@@ -0,0 +1,276 @@
+            return construct_json_resp_message(
+                "failed",
+                "worker registry contract instance is not initialized")
+
+    def worker_lookup_next(self, worker_type, org_id, application_id,
+                           lookup_tag):
+        if (self.__contract_instance is not None):
+            if not isinstance(worker_type, WorkerType):
+                logging.info("Invalid workerType {}".format(worker_type))
+                return construct_json_resp_message(
+                    "failed", "Invalid workerType {}".format(worker_type))
+            if not is_valid_hex_str(binascii.hexlify(org_id).decode("utf")):
+                logging.info("Invalid organization id {}".format(org_id))
+                return construct_json_resp_message(
+                    "failed", "Invalid organization id {}".format(org_id))
+            if not is_valid_hex_str(
+                    binascii.hexlify(application_id).decode("utf8")):
+                logging.info("Invalid application id {}".format(org_id))
+                return construct_json_resp_message(
+                    "failed", "Invalid application id {}".format(org_id))
+            lookupResult = self.__contract_instance.functions.workerLookUpNext(
+                worker_type.value,
+                org_id, application_id, lookup_tag).call()
+            return lookupResult
+        else:
+            logging.error(
+                "worker registry contract instance is not initialized")
+            return construct_json_resp_message("failed", "worker registry contract instance \
+                is not initialized")
+
+    def __validate(self, config):
+        """
+        validates parameter from config parameters for existence.
+        Returns false if validation fails and true if it success
+        """
+        if config["ethereum"]["worker_registry_contract_file"] is None:
+            logging.error("Missing worker registry contract file path!!")
+            return False
+        if config["ethereum"]["worker_registry_contract_address"] is None:
+            logging.error("Missing worker registry contract address!!")
+            return False
+        return True
+
+    def __initialize(self, config):
+        """
+        Initialize the parameters from config to instance variables.
+        """
+        self.__eth_client = EthereumWrapper(config)
+        tcf_home = environ.get("TCF_HOME", "../../../")
+        contract_file_name = tcf_home + "/" + \
+            config["ethereum"]["worker_registry_contract_file"]
+        contract_address = \
+            config["ethereum"]["worker_registry_contract_address"]
+        self.__contract_instance = self.__eth_client.get_contract_instance(
+            contract_file_name, contract_address
+        )
+
+    def worker_register(
+            self, worker_id, worker_type, org_id, application_ids, details):
+        """
+        Register new worker with details of worker
```

Comment:
Also mention the return value of the function. This function return status response formatted as JSON string.

Reply:
Changed to return None or error and on success return txn receipt.